test/distributed/org/apache/cassandra/distributed/test/JMXSslConfigDistributedTest.java

```diff
@@ -0,0 +1,202 @@
+        // uses it when Client SSL Socketfactory is initialized even if we don't need it here.
+        // The same default SSLContext.getDefault() will be used by other methods like testSystemSettings() in this test
+        // for the Server SSL Socketfactory and at that time we will need the keystore to be available
+        // All of the above is the issue because we run everything (JMX Server, Client) in the same JVM, multiple times
+        // and the SSLContext.getDefault() relies on static initialization that is reused
+        try (WithProperties ignored = new WithProperties().with("javax.net.ssl.trustStore", (String) validKeystore.get("truststore"),
+                                                                "javax.net.ssl.trustStorePassword", (String) validKeystore.get("truststore_password"),
+                                                                "javax.net.ssl.keyStore", (String) validKeystore.get("keystore"),
+                                                                "javax.net.ssl.keyStorePassword", (String) validKeystore.get("keystore_password"))
+        )
+        {
+            ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
+                                                                            .put("enabled", true)
+                                                                            .put("accepted_protocols", Arrays.asList("TLSv1.2", "TLSv1.3", "TLSv1.1"))
+                                                                            .build();
+
+            try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+                c.with(Feature.JMX).set("jmx_encryption_options", encryptionOptionsMap);
+            }).start())
+            {
+                Map<String, Object> jmxEnv = new HashMap<>();
+                configureClientSocketFactory(jmxEnv, encryptionOptionsMap);
+                // Invoke the same code vs duplicating any code from the JMXGetterCheckTest
+                JMXGetterCheckTest.testAllValidGetters(cluster, jmxEnv);
+            }
+        }
+    }
+
+    @Test
+    public void testClientAuth() throws Throwable
+    {
+        try (WithProperties ignored = new WithProperties().with("javax.net.ssl.trustStore", (String) validKeystore.get("truststore"),
+                                                                "javax.net.ssl.trustStorePassword", (String) validKeystore.get("truststore_password"),
+                                                                "javax.net.ssl.keyStore", (String) validKeystore.get("keystore"),
+                                                                "javax.net.ssl.keyStorePassword", (String) validKeystore.get("keystore_password"))
+        )
+        {
+            ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder().putAll(validKeystore)
```

Comment:
same

Reply:
same challenge here as mentioned before. I need to use the map variable later.